Watchman.Configuration/Generic/ResourceThresholds.cs

```diff
@@ -15,5 +18,30 @@ public override string ToString()
         {
             return Name ?? Pattern;
         }
+
+        public ResourceThresholds<TConfig> AsNamed(string name)
```

Comment:
moved this here so it's in one place. as I added the field then spent a while figuring out why it didn't work due to missing the mapping